test/string_test.rb

```diff
@@ -85,6 +87,20 @@
     end
   end
 
+  describe '#classified' do
+    it 'correctly recognizes a classified String' do
+      Hanami::Utils::String.new('hanami').classified?.must_equal(false)
+      Hanami::Utils::String.new('Hanami_router').classified?.must_equal(true)
+      Hanami::Utils::String.new('hanami-router').classified?.must_equal(false)
+      Hanami::Utils::String.new('Hanami/router').classified?.must_equal(false)
+      Hanami::Utils::String.new('Hanami/router').classified?.must_equal(false)
+      Hanami::Utils::String.new('Hanami/Router').classified?.must_equal(false)
+      Hanami::Utils::String.new('HanamiRouter').classified?.must_equal(true)
+      Hanami::Utils::String.new('AwesomeProject::Namespace').classified?.must_equal(true)
+      Hanami::Utils::String.new('AwesomeProject::SubNamespace').classified?.must_equal(true)
```

Comment:
Really strange logic for `Hanami_router` and `Hanami/Router`, but it just my opinion 😃